OneSignalSDK/onesignal/core/src/main/java/com/onesignal/common/services/ServiceProvider.kt

```diff
@@ -80,6 +85,26 @@ class ServiceProvider(
         }
     }
 
+    fun scheduleStartServices() {
```

Comment:
This method is stretching what I think this ServiceProvider should do. We already have a `StartupService`start()` method that I think this is a much better fit for. Can we not just change the code in that function instead?

Reply:
This implementation can replace what we have in `startupService.start()`. We can then also remove `IStartableService` from it's constructor, to avoid the main thread instantiation you are talking about.

Reply:
StartupService has been reworked so it no longer accepts lists of `IStartableService` and `IBootstrapService`. Instead, it will take a service provider and implement logic for 'bootstrap' and 'scheduleStart'.